test/ng/directive/ngModelSpec.js

```diff
@@ -847,6 +847,38 @@ describe('ngModel', function() {
         expect(ctrl.$valid).toBe(true);
       });
 
+      it('should treat all responses as boolean for synchronous validators', function() {
+        var curry = function(v) {
```

Comment:
You don't need a new function for this. You can use the internal helper: `valueFn()`